langlib/langlib-test/src/test/resources/test-src/valuelib_test.bal

```diff
@@ -1892,6 +1892,16 @@ function testToStringOnCycles() {
      assert(x.toString(), "{\"ee\":3,\"1\":{\"qq\":5,\"1\":...,\"2\":[2,3,5,...]}}");
 }
 
+function testToJsonWithCyclicParameter() {
+    anydata[] x = [];
+    x.push(x);
+    json|error y = trap x.toJson();
+    assert(y is error, true);
+    error err = <error> y;
+    assert(err.message(), "{ballerina/lang.value}CyclicValueReferenceError");
+    assert(<string> checkpanic err.detail()["message"], "'anydata[]' value has cyclic reference");
```

Comment:
you can do the following without casting:
```suggestion
    assert(err.detail()["message"].toString(), "'anydata[]' value has cyclic reference");
```

Reply:
It gave compilation errors. Since `err.detail()["message"]` is `string|error` we can't directly use `.toString()` on it right?